janus_server/src/hpke.rs

```diff
@@ -144,38 +144,166 @@ impl HpkeSender {
         plaintext: &[u8],
         associated_data: &[u8],
     ) -> Result<HpkeCiphertext, Error> {
+        let application_info = HpkeApplicationInfo::new(
+            self.task_id,
+            self.label,
+            self.sender_role,
+            self.recipient_role,
+        );
         // We must manually dispatch to each possible specialization of seal
         match (
             self.recipient_config.kem_id,
             self.recipient_config.kdf_id,
             self.recipient_config.aead_id,
         ) {
+            (HpkeKemId::P256HkdfSha256, HpkeKdfId::HkdfSha256, HpkeAeadId::Aes128Gcm) => {
```

Comment:
We could reduce repetition by having each `match` arm return the function to be called -- since every arm calls a function with the same signature we can put it in a variable, then call the variable.

That is, I replaced the body with the following in my pre-this-PR copy and it worked:
```
        // We must manually dispatch to each possible specialization of seal
        let seal = match (
            self.recipient_config.kem_id,
            self.recipient_config.kdf_id,
            self.recipient_config.aead_id,
        ) {
            (HpkeKemId::P256HkdfSha256, HpkeKdfId::HkdfSha256, HpkeAeadId::Aes256Gcm) => {
                seal::<AesGcm256, HkdfSha256, DhP256HkdfSha256>
            }
            (HpkeKemId::X25519HkdfSha256, HpkeKdfId::HkdfSha512, HpkeAeadId::ChaCha20Poly1305) => {
                seal::<ChaCha20Poly1305, HkdfSha512, X25519HkdfSha256>
            }
            (_, _, _) => {
                return Err(Error::InvalidConfiguration(
                    "unsupported set of HPKE algorithms",
                ))
            }
        };
        seal(
            &self.recipient_config,
            HpkeApplicationInfo::new(
                self.task_id,
                self.label,
                self.sender_role,
                self.recipient_role,
            ),
            plaintext,
            associated_data,
        )
```